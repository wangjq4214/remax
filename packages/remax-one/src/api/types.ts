export interface NavigateToParams {
  url: string;
}

export declare function navigateTo(params: NavigateToParams): Promise<void>;

export interface NavigateBackParams {
  /** 返回的页面数 */
  delta?: number;
}

export declare function navigateBack(params?: NavigateBackParams): Promise<void>;

export interface RedirectToParams {
  url: string;
}

export declare function redirectTo(params: RedirectToParams): Promise<void>;

export interface ReLaunchParams {
  url: string;
}

export declare function reLaunch(params: ReLaunchParams): Promise<void>;

export interface SwitchTabParams {
  url: string;
}

export declare function switchTab(params: SwitchTabParams): Promise<void>;
