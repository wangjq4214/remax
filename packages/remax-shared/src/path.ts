export function slash(path: string) {
  return /^\\\\\?\\/.test(path) ? path : path.replace(/\\/g, `/`);
}
