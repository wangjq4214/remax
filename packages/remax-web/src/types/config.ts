interface TabItem {
  /**
   * 设置页面路径
   */
  url: string;
  /**
   * 名称
   */
  title: string;
  /**
   * 平常图标路径
   */
  image: string;
  /**
   * 高亮图标路径
   */
  activeImage?: string;
}

/** 全局配置文件 */
export interface AppConfig {
  /**
   * 设置页面路径
   */
  pages: string[];
  /**
   * 页面默认标题
   */
  title: string;
  /**
   * 默认值：false
   * 是否开启全局的下拉刷新。
   */
  pullToRefresh?: boolean;
  /**
   * 默认值：50
   * 页面触底事件触发时距页面底部距离，单位为 px。
   */
  reachBottomOffset?: number;
  /**
   * 默认值：hash
   * history 类型
   */
  historyType?: 'hash' | 'browser';
  /**
   * 设置底部 tab bar 的表现
   */
  tabBar?: {
    /**
     * 标题颜色
     */
    titleColor?: string;
    /**
     * 选中标题颜色
     */
    activeTitleColor?: string;
    /**
     * 背景色
     */
    backgroundColor?: string;
    /**
     * 每个 tab 配置
     */
    items: TabItem[];
  };
}

/** 页面配置文件 */
export interface PageConfig {
  /**
   * 页面默认标题
   */
  title: string;
  /**
   * 默认值：false
   * 是否开启全局的下拉刷新。
   */
  pullToRefresh?: boolean;
  /**
   * 默认值：50
   * 页面触底事件触发时距页面底部距离，单位为 px。
   */
  reachBottomOffset?: number;
}
