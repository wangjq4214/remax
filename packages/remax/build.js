module.exports.buildApp = require('@remax/cli/lib/build').build;
