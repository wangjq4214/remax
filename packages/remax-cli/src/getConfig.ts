import fs from 'fs';
import path from 'path';
import { getDefaultOptions } from './defaultOptions';
import { Options } from '@remax/types';
import validateOptions from 'schema-utils';
import schema from './OptionsSchema.json';

export interface CliOptions {
  target: string;
}

function readJavascriptConfig(path: string) {
  delete require.cache[path];
  const config = require(path);
  return config || {};
}

export default function getConfig(validate = true): Options {
  const configPath: string = path.join(process.cwd(), './remax.config');

  let options = {};

  if (fs.existsSync(configPath + '.js')) {
    options = readJavascriptConfig(configPath + '.js');
  }

  if (validate) {
    validateOptions(schema as any, options, {
      name: 'remax',
    });
  }

  return {
    ...getDefaultOptions(),
    ...options,
  };
}
