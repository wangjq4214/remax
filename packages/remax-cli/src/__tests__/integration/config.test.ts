import * as path from 'path';
import build from './helpers/build';
import getConfig from '../../getConfig';
import { Platform } from '@remax/types';
import runTest from './helpers/runTest';

describe('remax config', () => {
  it('override output', () => {
    process.chdir(path.join(__dirname, 'fixtures/config'));
    const result = getConfig();
    expect(result.output).toEqual('build');
  });

  it('schema validation', () => {
    process.chdir(path.join(__dirname, 'fixtures/exception/remax.config'));

    expect(() => {
      getConfig();
    }).toThrowErrorMatchingSnapshot();
  });

  it('throw error when missing pages config in app.config', async () => {
    await expect(build('exception', Platform.ali)).rejects.toThrow('app.config.js|ts 并未配置页面参数');
  });

  runTest('config-add-css-rule');
});
