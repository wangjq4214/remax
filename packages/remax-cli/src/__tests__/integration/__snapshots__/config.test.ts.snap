// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remax config schema validation 1`] = `
"Invalid configuration object. remax has been initialized using a configuration object that does not match the API schema.
 - configuration has an unknown property 'xxx'. These properties are valid:
   object { turboPages?, pxToRpx?, cwd?, progress?, compressTemplate?, output?, rootDir?, UNSAFE_wechatTemplateDepth?, one?, configWebpack?, plugins? }
 - configuration.turboPages should be an array:
   [any, ...]
 - configuration.pxToRpx should be a boolean.
 - configuration.cwd should be a string.
 - configuration.progress should be a boolean.
 - configuration.compressTemplate should be a boolean.
 - configuration.output should be a string.
 - configuration.rootDir should be a string.
 - configuration.UNSAFE_wechatTemplateDepth should be one of these:
   object { … } | number
   Details:
    * configuration.UNSAFE_wechatTemplateDepth should be an object:
      object { … }
    * configuration.UNSAFE_wechatTemplateDepth should be a number.
 - configuration.configWebpack should be an instance of function.
 - configuration.plugins should be an array:
   [any, ...]"
`;
